packages/solid-query/src/createMutationState.ts

```diff
@@ -0,0 +1,59 @@
+import { type Accessor, createMemo, onCleanup } from "solid-js";
+import {
+    type DefaultError,
+    type Mutation,
+    type MutationCache,
+    type MutationFilters,
+    type MutationState,
+    type QueryClient,
+    replaceEqualDeep,
+    useQueryClient,
+} from "."
+
+type MutationStateOptions<TResult = MutationState> = {
+    filters?: MutationFilters
+    select?: (
+        mutation: Mutation<unknown, DefaultError, unknown, unknown>,
+    ) => TResult
+};
+
+function getResult<TResult = MutationState>(
+    mutationCache: MutationCache,
+    options: MutationStateOptions<TResult>,
+): Array<TResult> {
+    return mutationCache
+        .findAll(options.filters)
+        .map(
+            (mutation): TResult =>
+                (options.select
+                    ? options.select(
+                        mutation as Mutation<unknown, DefaultError, unknown, unknown>,
+                    )
+                    : mutation.state) as TResult,
+        )
+}
+
+export function createMutationState<TResult = MutationState>(
```

Comment:
Thanks for working on this! This is looking great! A few things. We should still call this hook `useMutationState` since we're still using data from already created state. Similar to how `useIsMutating` is named!

Reply:
Made the suggested changes. Still having some trouble with the pushing to variables and the following

```
expect(variables).toEqual([[], [1], []])
```

Reply:
I think I got them passing? Added a createEffect